### PR TITLE
adjust for lineHeight and letterSpacing in webgl custom glyph rendering

### LINKED ISSUE #3427: Letter spacing and line height is off for custom glyphs under webgl
image
image

### PR DESCRIPTION
**Pull request by meganrogge**
fix #3427

### CODE REVIEW

File: addons/xterm-addon-webgl/src/atlas/WebglCharAtlas.ts [hunk 1]
```diff
@@ -567,8 +567,8 @@ export class WebglCharAtlas implements IDisposable {
         y: (boundingBox.bottom - boundingBox.top + 1) / TEXTURE_HEIGHT
       },
       offset: {
-        x: -boundingBox.left + (restrictedGlyph ? 0 : TMP_CANVAS_GLYPH_PADDING),
-        y: -boundingBox.top + (restrictedGlyph ? 0 : TMP_CANVAS_GLYPH_PADDING)
+        x: -boundingBox.left + (restrictedGlyph ? 0 : TMP_CANVAS_GLYPH_PADDING) + (customGlyph ? Math.floor(this._config.letterSpacing / 2) : 0),
+        y: -boundingBox.top + (restrictedGlyph ? 0 : TMP_CANVAS_GLYPH_PADDING) + (customGlyph ? this._config.lineHeight === 1 ? 0 : Math.round((this._config.scaledCellHeight - this._config.scaledCharHeight) / 2) : 0)
```
Comment (meganrogge) [inline addons/xterm-addon-webgl/src/atlas/WebglCharAtlas.ts:571]: pretty ugly to put these inline, but thought creating consts might be costly?

Comment (jerch) [reply addons/xterm-addon-webgl/src/atlas/WebglCharAtlas.ts:571]: If perf is a concern - wouldn't it be better to build the box glyphs in the atlas already with those spacing/scaledHeight deps precalculated? Would avoid any special casing here (always hits hard in hot paths), and I think changes to those values are rare, thus it should be no biggie to rebuild dependent glyphs in the atlas.

Comment (jerch) [reply addons/xterm-addon-webgl/src/atlas/WebglCharAtlas.ts:571]: Yepp, was meant as a cache warm-up. But I have no clue of the bigger picture here, so I wasnt aware of the color issue :sweat_smile:

**Review by Tyriar — APPROVED** [review]
The optimization is probably overkill, can always come back and improve later

**Pull request merged**